compiler/natives/os/os.go

```diff
@@ -24,3 +24,27 @@ func init() {
 }
 
 func runtime_beforeExit() {}
+
+// Override the builtin os.Hostname(). The builtin Hostname() will result in
+// a non-working syscall on most (all?) platforms
+func Hostname() (string, error) {
+	// Browser
+	// The browser doesn't know anything about any hostname, so we use the host part of
+	// the current url.
+	if location := js.Global.Get("location"); location != js.Undefined {
+		if browserHostname := location.Get("hostname"); browserHostname != js.Undefined {
+			return browserHostname.String(), nil
+		}
+	}
+
+	// Node.js
+	// We call node's require("os").hostname() which should return current hostname
+	if require := js.Global.Get("require"); require != js.Undefined {
+		// "os" is in nodejs core, it should always exist
+		if nodeHostname := require.Invoke("os").Call("hostname"); nodeHostname != js.Undefined {
```

Comment:
I'm not sure what the right thing to do is. We cannot return the correct `client` hostname in the browser no matter what we do.

I just wan't the warning to go away. An almost perfect implementation for me would be:
```go
// Make @abrander stop complaining :)
func Hostname() (string, error) {
	return "not-supported.example.com", nil
}
```

The current implementation is the most correct in browser land, but there's no reason that it shouldn't work as expected in Node land.